File: backend/app/Models/ChurchBankAccount.php
<?php

namespace App\Models;

use Database\Factories\ChurchBankAccountFactory;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

#[Fillable([
    'bank_name',
    'account_number',
    'account_holder_name',
    'is_active',
    'display_order',
])]
class ChurchBankAccount extends Model
{
    /** @use HasFactory<ChurchBankAccountFactory> */
    use HasFactory;

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
            'display_order' => 'integer',
        ];
    }

    /**
     * Scope a query to only include active church bank accounts.
     *
     * @param  Builder<self>  $query
     * @return Builder<self>
     */
    public function scopeActive(Builder $query): Builder
    {
        return $query->where('is_active', true)->orderBy('display_order');
    }
}

File: mobile/lib/features/donations/domain/entities/church_bank_account.dart
class ChurchBankAccount {
  final int id;
  final String bankName;
  final String accountNumber;
  final String accountHolderName;
  final bool isActive;
  final int displayOrder;

  const ChurchBankAccount({
    required this.id,
    required this.bankName,
    required this.accountNumber,
    required this.accountHolderName,
    required this.isActive,
    required this.displayOrder,
  });
}
